src/components/CatalogPage/CatalogPage.tsx

```diff
@@ -0,0 +1,190 @@
+
+  const [products, setProducts] = useState<ProductProjectionResponse>({
+    productProjectionSearch: {
+      limit: 0,
+      count: 0,
+      offset: 0,
+      results: [],
+    },
+  });
+
+  const [currentRangeValue, setCurrentRangeValue] = useState<number[]>([
+    0, 3100,
+  ]);
+  const handleRangeSlider = (event: number | number[]) => {
+    if (typeof event !== 'number') {
+      setCurrentRangeValue(event);
+    }
+  };
+
+  const handleCategories = (event: React.ChangeEvent) => {
+    const target = event.target as HTMLInputElement;
+    if (target.checked) {
+      setCurrentCategories([...currentCategories, target.value]);
+    } else {
+      currentCategories.splice(currentCategories.indexOf(target.value), 1);
+      setCurrentCategories([...currentCategories]);
+    }
+  };
+
+  const handleCurrentSort = (event: React.ChangeEvent) => {
+    const target = event.target as HTMLInputElement;
+    setCurrentSort(target.value);
+  };
+
+  const handleCurrentSearch = (event: FormEvent) => {
+    const target = event.target as HTMLInputElement;
+    setcurrentSearch(target.value);
+  };
+
+  useEffect(() => {
+    handleLoading(true);
+    const productService = new ProductService();
+    if (
+      currentSort ||
+      currentSearch ||
+      currentCategories.length > 0 ||
+      currentRangeValue[0] !== 0 ||
+      currentRangeValue[1] !== 3100
+    ) {
+      const data: Promise<CTResponse> = productService.getProductsWithFilters(
+        'EN-GB',
+        currentRangeValue,
+        currentSort,
+        currentSearch,
+        currentCategories
+      );
+      data.then((response) => {
+        handleLoading(false);
+        setProducts(
+          (response.data as CTResponse).data as ProductProjectionResponse
+        );
+      });
+    } else {
+      const data: Promise<CTResponse> = productService.getProductsAll();
+      data.then((response) => {
+        handleLoading(false);
+        setProducts(
+          (response.data as CTResponse).data as ProductProjectionResponse
+        );
+      });
+    }
+  }, [currentSort, currentSearch, currentRangeValue, currentCategories]);
+
+  return (
+    <section className="flex relative">
+      <div>
+        <FilterSection
+          handleCategories={handleCategories}
+          currentRangeValue={currentRangeValue}
+          handleRangeSlider={handleRangeSlider}
+          showFilter={showFilter}
+          setShowFilter={setShowFilter}
+        />
+      </div>
+      <div className="flex gap-5 flex-col w-full p-5">
+        <SortSection
+          handleCurrentSearch={handleCurrentSearch}
+          handleCurrentSort={handleCurrentSort}
+          showFilter={showFilter}
+          setShowFilter={setShowFilter}
+        />
+        <div className="flex flex-col gap-3">
+          <h1 className="text-2xl">Products</h1>
+          <section className="flex flex-col gap-5 flex-wrap">
+            {products.productProjectionSearch.results.length > 0 ? (
```

Comment:
combine condition and use more simplier logic


`    {products.productProjectionSearch.results.length > 0 ? (
              products.productProjectionSearch.results.map((product) => {
                const { key, masterVariant, categories, description, name } = product;
                const { prices, images } = masterVariant;
                const basePrice = convertPrice(
                  prices?.[0]?.value.centAmount || 0,
                  prices?.[0]?.value.fractionDigits || 0
                );
                const discountedPrice = prices?.[0]?.discounted
                  ? convertPrice(
                      prices[0].discounted.value.centAmount,
                      prices[0].discounted.value.fractionDigits
                    )
                  : basePrice;

                return (
                  <ProductPreviewItem
                    key={key}
                    id={key || ''}
                    imgUrl={images?.[0]?.url || ''}
                    productCategory={categories.length ? categories.map((cat) => (cat as unknown as Category).name).join(', ') : 'no category'}
                    productDescription={description || 'no description'}
                    productName={name || ''}
                    productPrice={discountedPrice}
                    productOldPrice={prices?.[0]?.discounted ? basePrice : ''}
                  />
                );
              })
            ) : (
              <p>No match</p>
            )}`